src/components/registerForm.vue

```diff
@@ -0,0 +1,124 @@
+<template>
+    <div class="mx-auto p-5 rounded-3xl w-min" style="background-color: #313030;">
+        <div class="p-5">
+            <h1>Register</h1>
+            <p>Fill all fields</p>
+            <div class="mt-12">
+                <form>
+                    <p v-if="parsedMessage" class="text-red-600 w-full mb-4 text-xs">{{ parsedMessage }}</p>
+                    <div class="input-div">
+                        <input class="w-full darkcolor lightbg mb-6 p-1.5 rounded-xl" type="text" placeholder="Name" v-model="name" />
+                    </div>
+                    <div class="input-div">
+                        <input class="w-full darkcolor lightbg mb-6 p-1.5 rounded-xl" type="email" placeholder="E-mail" v-model="email" />
+                    </div>
+                    <div class="input-div">
+                        <input class="w-full darkcolor lightbg mb-6 p-1.5 rounded-xl" type="password" placeholder="Password" v-model="password" />
+                    </div>
+                    <div class="input-div">
+                        <input class="w-full darkcolor lightbg mb-6 p-1.5 rounded-xl" type="password" placeholder="Confirm Password" v-model="confirmPassword" />
+                    </div>
+                    <div class="mt-6">
+                        <button type="submit" class="w-full darkbg rounded-xl p-2" @click.prevent="register">Register</button>
+                    </div>
+                </form>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+export default {
+    data() {
+        return {
+            name: "",
+            email: "",
+            password: "",
+            confirmPassword: "",
+            errorMessage: "",
+            validPassword: false,
+            passwordsMatch: false
+        }
+    },
+    props: {
+        endpoint: {
+            type: String,
+            required: true,
+        }
+    },
+    computed: {
+        parsedMessage(): String {
+            return this.errorMessage;
+        }
+    },
+    watch: {
+        password() {
+            const length = this.password.length > 8;
+            const containsUppercase = /[A-Z]/.test(this.password);
+            const containsLowercase = /[a-z]/.test(this.password);
+            const containsNumber = /[0-9]/.test(this.password);
+            const containsSpecial = /[#?!@$%^&*-]/.test(this.password);
+            this.validPassword = length && containsUppercase && containsLowercase && containsNumber && containsSpecial;
```

Comment:
Should be handled by backend, just remove and we will fix later

Reply:
Changes added